feat: take tick range from URL

dmihal opened the pull request:
Lets an outside application pre-set the price range, for example:

`/add/ETH/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/3000?minPrice=100&maxPrice=500`

dmihal commented:
Working test link:
https://interface-git-fork-dmihal-tick-range-uniswap.vercel.app/#/add/ETH/0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48/3000?chain=mainnet&minPrice=2000&maxPrice=3000

JFrankfurt commented:
I love this. @tinaszheng what do you think of the behavior here? @dmihal has the effect removing the param instead of maintaining state sync w/ the ui like we do for fee level/asset/network. My gut says we should sync as the user changes so the url is shareable into a specific range.

moodysalem commented on src/pages/AddLiquidity/index.tsx:
```diff
@@ -150,6 +150,31 @@ export default function AddLiquidity({
 
+  useEffect(() => {
+    if (history.location.search) {
+      const search = new URLSearchParams(history.location.search)
```
we have a `useParsedQueryString()` hook

dmihal replied:
Cool, I've switched to using it

moodysalem commented on src/pages/AddLiquidity/index.tsx:
```diff
@@ -150,6 +150,31 @@ export default function AddLiquidity({
+      }
+
+      if (updated) {
+        history.replace({ search: search.toString() })
```
it seems like this would cause an infinite loop?

dmihal replied:
Updated, query params now stay in the URL

tinaszheng commented:
ooo thanks so much for your contribution @dmihal !! looks great but i do agree that we should probably persist the query params and also sync the URL with the app ranges. you can take a look at https://github.com/Uniswap/interface/blob/main/src/components/Header/NetworkSelector.tsx#L284 for some inspiration for how we sync network params with the app (although that one is probably more complicated)

dmihal commented:
Hey everyone, thanks for the review!

I've updated the PR to persist the URL param, so users can easily have "sharable" URLs, similar to the network param shared by @tinaszheng

I've also added  useParsedQueryString at @moodysalem's suggestion.

tinaszheng commented on src/pages/AddLiquidity/index.tsx:
```diff
@@ -150,6 +153,16 @@ export default function AddLiquidity({
+      onLeftRangeInput(parsedQs.minPrice)
+    }
+
+    if (typeof parsedQs.maxPrice === 'string' && parsedQs.minPrice !== rightRangeTypedValue) {
```
i believe you want `parsedQs.maxPrice` here?

dmihal replied:
Good catch!

tinaszheng reviewed (changes requested):
just 2 requested changes - otherwise looks good so far!

JFrankfurt reviewed (approved):
lgtm -- I noticed though that some ticks end up with a bunch of trailing 0's, while others do not. (i.e., `40.100000` and `60.2`)

JFrankfurt commented:
thanks @dmihal !

dmihal commented:
Thanks for merging this @JFrankfurt & @tinaszheng!

Great timing too, we just had a team call today, and hopefully launching the staker soon!
